lib/gui/components/update-notifier.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @module Etcher.Components.UpdateNotifier
+ */
+
+const angular = require('angular');
+const electron = require('electron');
+const Bluebird = require('bluebird');
+const _ = require('lodash');
+const settings = require('../models/settings');
+const units = require('../../shared/units');
+const release = require('../../shared/release');
+const packageJSON = require('../../../package.json');
+
+const MODULE_NAME = 'Etcher.Components.UpdateNotifier';
+const UpdateNotifier = angular.module(MODULE_NAME, [
+  require('../os/open-external/open-external'),
+  require('../modules/analytics')
+]);
+
+UpdateNotifier.service('UpdateNotifierService', function(AnalyticsService, OSOpenExternalService) {
+
+  /**
+   * @summary The number of days the update notifier can be put to sleep
+   * @constant
+   * @private
+   * @type {Number}
+   */
+  this.UPDATE_NOTIFIER_SLEEP_DAYS = packageJSON.updates.sleepDays;
+
+  /**
+   * @summary Determine if it's time to check for updates
+   * @function
+   * @public
+   *
+   * @param {Object} [options] - options
```

Comment:
Given that `options.releaseType` isn't in square brackets (indicating that it's non-optional?), should `options` here also not be in square brackets?